compiler/plugins/target/AMD-AIE/aie/CMakeLists.txt

```diff
@@ -304,6 +304,8 @@ iree_cc_library(
     "AIEObjectFifoStatefulTransform.cpp"
     "AIEDmaToNpu.cpp"
     "AIEXToStandard.cpp"
+    # Passes needed by AIR-AIE lowering
+    "${IREE_MLIR_AIE_SOURCE_DIR}/lib/Dialect/AIE/Transforms/AIENormalizeAddressSpaces.cpp"
```

Comment:
Where did you find you needed this pass? I couldn't find any tests that failed without it? Anyway can you bring it into our source tree like the others.

Reply:
This is needed in  the llvm conversion, otherwise the buffers have address spaces on them which peano doesnt like, will in tree this.

Reply:
is this one okay to use from in-tree with you though?

Reply:
Well I left this comment before understanding what you're doing so the answer is no (not because it's okay/not-okay but for the reason I give below). But also I don't actually think you need this? I tested extensively (included E2E with numerics) and I never needed this pass (which does nothing but set a memory address space attribute which is never then queried/used in a subsequent analysis).

Reply:
Hmm, in my testing, peano was failing to lower to assembly code without this pass due to the memory addressspace difference when not using this pass. That being said my peano is I believe 3 months old (didnt update it since it wasnt giving me issues) so once I get the CI green, I will remove this pass and if it stays green after then that means it was my old peano issue.

Reply:
Ya, its happening on CI, so I will keep the pass, open to opinions on where you want it for now though
https://github.com/nod-ai/iree-amd-aie/actions/runs/9686354878/job/26729179745#step:6:268

Reply:
Just if you can do what I originally proposed that would be good (just copy the file to the same place as the rest of the things and update the header). But I'll just reiterate: if you actually want to make this happen then you should not use those same passes because I will delete/coalesce/condense them very soon (next week when I get back).

Reply:
I am confused, you want me to move the AIENormalizeAddressSpaces passes with the rest of your passes but then you are saying you may delete it when you refactor? I do want this path to be stable..

Reply:
I made my own library with the "upstream" mlir-aie passes that includes the AIENormalizeAddressSpaces pass.

Reply:
> I am confused, you want me to move the AIENormalizeAddressSpaces passes with the rest of your passes but then you are saying you may delete it when you refactor?

Well I was saying that 

1. If you want to continue to use my passes then yes move over `AIENormalizeAddressSpaces`
2. But don't use my passes because they'll go away

> I made my own library with the "upstream" mlir-aie passes that includes the AIENormalizeAddressSpaces pass.

Yup that's good (I'll move over `AIENormalizeAddressSpaces` to my dir myself soon).

Reply:
Cool, thanks, makes sense!